amr-wind/turbulence/LES/OneEqKsgs.cpp

```diff
@@ -0,0 +1,219 @@
+#include "amr-wind/turbulence/LES/OneEqKsgs.H"
+#include "amr-wind/equation_systems/PDEBase.H"
+#include "amr-wind/turbulence/TurbModelDefs.H"
+#include "amr-wind/derive/derive_K.H"
+#include "amr-wind/turbulence/turb_utils.H"
+#include "amr-wind/equation_systems/tke/TKE.H"
+
+#include "AMReX_ParmParse.H"
+
+namespace amr_wind {
+namespace turbulence {
+
+template <typename Transport>
+OneEqKsgs<Transport>::OneEqKsgs(CFDSim& sim)
+    : TurbModelBase<Transport>(sim), m_vel(sim.repo().get_field("velocity")),
+      m_turb_lscale(sim.repo().declare_field("turb_lscale",1, 1, 1)),
+      m_shear_prod(sim.repo().declare_field("shear_prod",1, 1, 1)),
+      m_buoy_prod(sim.repo().declare_field("buoy_prod",1, 1, 1)),
+      m_rho(sim.repo().get_field("density"))
+{
+    auto& tke_eqn = sim.pde_manager().register_transport_pde(pde::TKE::pde_name());
+    m_tke = &(tke_eqn.fields().field);
+
+    //Turbulent length scale field
+    this->m_sim.io_manager().register_io_var("turb_lscale");
+    this->m_sim.io_manager().register_io_var("shear_prod");
+    this->m_sim.io_manager().register_io_var("buoy_prod");
+    
+}
+
+template <typename Transport>
+OneEqKsgs<Transport>::~OneEqKsgs() = default;
+
+template <typename Transport>
+OneEqKsgsM84<Transport>::OneEqKsgsM84(CFDSim& sim)
+    : OneEqKsgs<Transport>(sim),
+      m_temperature(sim.repo().get_field("temperature"))
+{
+
+    auto& phy_mgr = this->m_sim.physics_manager();
+    if (!phy_mgr.contains("ABL")) 
+        amrex::Abort("OneEqKsgsM84 model only works with ABL physics");
+        
+    {
+        const std::string coeffs_dict = this->model_name() + "_coeffs";
+        amrex::ParmParse pp(coeffs_dict);
+        pp.query("Ceps", this->m_Ceps);
+        pp.query("Ce", this->m_Ce);
+    }
+
+    {
+        amrex::ParmParse pp("ABL");
+        pp.get("reference_temperature", m_ref_theta);
+    }
```

Comment:
Currently there is no `reference_temperature` in ABL namespace, `reference_temperature` is only available if the user has activated `BoussinesqBouyancy`, so now we have to specify `reference_temperature`. This is not ideal.

Reply:
I'd prefer to have reference temperature in the ABL namespace as it enters several places related to the ABL including Boussinesq Buoyancy, turbulence model and wall model. Could we discuss a change related to this in the input file?

Reply:
I think having `reference_temperature` in `ABL` or better yet some kind of `reference_state` namespace is a good idea.

Reply:
Ok. I'll keep it in the ABL namespace. Should I modify the Boussinesq Buoyancy as a part of this pull request then>

Reply:
Yes. Can you update the BoussinesqBuoyancy behavior so that we only have one reference temperature specified?